core-bundle/src/Resources/contao/widgets/SerpPreview.php

```diff
@@ -72,115 +90,116 @@ public function generate()
 
 	private function getTitle(Model $model)
 	{
-		if (!isset($this->serpPreview['title']))
+		if (!isset($this->titleFields))
 		{
 			return $model->title;
 		}
 
-		if (\is_array($this->serpPreview['title']))
-		{
-			return $model->{$this->serpPreview['title'][0]} ?: $model->{$this->serpPreview['title'][1]};
-		}
-
-		return $model->{$this->serpPreview['title']};
+		return $model->{$this->titleFields[0]} ?: $model->{$this->titleFields[1]};
 	}
 
 	private function getDescription(Model $model)
 	{
-		if (!isset($this->serpPreview['description']))
+		if (!isset($this->descriptionFields))
 		{
 			return $model->description;
 		}
 
-		if (\is_array($this->serpPreview['description']))
+		return $model->{$this->descriptionFields[0]} ?: $model->{$this->descriptionFields[1]};
+	}
+
+	private function getAlias(Model $model)
+	{
+		if (!isset($this->aliasField))
 		{
-			return $model->{$this->serpPreview['description'][0]} ?: $model->{$this->serpPreview['description'][1]};
+			return $model->alias;
 		}
 
-		return $model->{$this->serpPreview['description']};
+		return $model->{$this->aliasField};
 	}
 
+	/**
+	 * @todo Use the router to generate the URL in a future version (see #831)
+	 */
 	private function getUrl(Model $model)
 	{
-		if (isset($this->serpPreview['url']))
+		if (!isset($this->url_callback))
 		{
-			return $this->serpPreview['url'];
+			throw new \LogicException('No url_callback given');
 		}
 
-		// FIXME: use the router to generate the URL (see #831)
-		switch (true)
-		{
-			case $model instanceof PageModel:
-				return $model->getAbsoluteUrl();
-
-			case $model instanceof NewsModel:
-				return News::generateNewsUrl($model, false, true);
+		$placeholder = bin2hex(random_bytes(10));
 
-			case $model instanceof CalendarEventsModel:
-				return Events::generateEventUrl($model, true);
+		// Pass a detached clone with the alias set to the placeholder
+		$tempModel = clone $model;
+		$tempModel->origAlias = $tempModel->alias;
+		$tempModel->alias = $placeholder;
```

Comment:
should probably use the `$aliasField` here?

Reply:
Fixed in 07cc4886e37d05be4b7646f27faa3859e448c98c.